### ISSUE TITLE
Can we build JEDI PIO modules on Cheyenne in Debug?

### ISSUE THREAD

**Issue by svahl991** [JCSDA/jedi-stack#134]
The mpas-jedi team has discovered that building the PIO library in Debug instead of Release (mostly) solves JCSDA/mpas-jedi#201, which was a major problem for us. So for all of our cycling experiments we have been using a custom Debug build of PIO and not using the JEDI PIO module on Cheyenne. This has had no noticeable affect on performance.

I wanted to ask if we could just build the JEDI PIO modules on Cheyenne in Debug so that we can go back to using those. Is PIO used anywhere else in JEDI other than the MPAS model?

Here is the line in the mpas-bundle build script where we build PIO, for reference. Notice the `PIO_ENABLE_TIMING` flag is turned off.
https://github.com/JCSDA/mpas-bundle/blob/963765863329974d447f26c0177bf30dddd380a9/build_mpas.csh#L368

**Comment by svahl991** [comment]
I thought I'd also mention here that PIO just had a release (2.5.1) last month:
https://github.com/NCAR/ParallelIO/releases

**Comment by mmiesch** [comment]
Sure, I can build a debug PIO module this afternoon.  Is this for gnu-openmpi?

**Comment by svahl991** [comment]
Yes, gnu-openmpi. (As we discussed over email last week, we've had problems with using intel.)

**Comment by mmiesch** [comment]
Ok - the module is available.  It is loaded by default with the `jedi/gnu-openmpi` module but you can also load it explicitly with:
```
module load pio-2.5.0-debug
```
I tried 2.5.1 but the git clone command failed - not known.  Are you sure it's available on github?  Does it have a different name?

**Comment by rickgrubin** [comment]
`https://github.com/NCAR/ParallelIO
`

This worked for me:

```
% git clone https://github.com/NCAR/ParallelIO.git
% head CMakeLists.txt
[...]
# The project version number.
set(VERSION_MAJOR   2   CACHE STRING "Project major version number.")
set(VERSION_MINOR   5   CACHE STRING "Project minor version number.")
set(VERSION_PATCH   1   CACHE STRING "Project patch version number.")
```

**Comment by svahl991** [comment]
Thanks! 

And getting version 2.5.1 instead of 2.5.0 is not important to us. I just thought I'd mention it.

**Comment by mmiesch** [comment]
> `https://github.com/NCAR/ParallelIO `
> 
> This worked for me:
> 
> ```
> % git clone https://github.com/NCAR/ParallelIO.git
> % head CMakeLists.txt
> [...]
> # The project version number.
> set(VERSION_MAJOR   2   CACHE STRING "Project major version number.")
> set(VERSION_MINOR   5   CACHE STRING "Project minor version number.")
> set(VERSION_PATCH   1   CACHE STRING "Project patch version number.")
> ```

Thanks @rickgrubin - I didn't look into it too closely - I just saw that blindly replacing the 0 with a 1 led to an error so apparently they haven't christened a `2_5_1` branch yet.  Or have they?

**Comment by mmiesch** [comment]
```bash
miesch@cheyenne5:~/jedi/jedi-stack/pkg/ParallelIO> git fetch
miesch@cheyenne5:~/jedi/jedi-stack/pkg/ParallelIO> git checkout pio2_5_1
error: pathspec 'pio2_5_1' did not match any file(s) known to git.
miesch@cheyenne5:~/jedi/jedi-stack/pkg/ParallelIO> git checkout pio2_5_0
HEAD is now at 751ff3d1... Merge pull request #1614 from NCAR/ejh_prepare
miesch@cheyenne5:~/jedi/jedi-stack/pkg/ParallelIO> 
```

Nothing to worry about though

**Comment by svahl991** [comment]
I think maybe they changed the format of the git tag, adding another underscore: pio_2_5_1
https://github.com/NCAR/ParallelIO/tree/pio_2_5_1
